### ISSUE TITLE
Can't create more than max_prepared_stmt_count statements

### ISSUE THREAD

**Issue by yiming33** [vert-x3/issues#444]
![image](https://user-images.githubusercontent.com/19584310/51723763-52a0da00-2095-11e9-80e6-fe3f4bfdcbe0.png)

In my process of using vertx-mysql-postgresql-client, I found a problem. Every time I use the prepared statement, mysql's Com_stmt_execute will be +1, but Com_stmt_close will not be +1, so there is a problem: when Com_stmt_execute is greater than the maximum After the value, the error in Figure 1 will appear. I hope to give a solution.

![image](https://user-images.githubusercontent.com/19584310/51723838-94ca1b80-2095-11e9-8059-aa2ff406f79c.png)

![image](https://user-images.githubusercontent.com/19584310/51724101-bc6db380-2096-11e9-820b-dd020a02eb7f.png)

The problem is urgent, I hope I can give me some help, thank you very much.

**Comment by yiming33** [comment]
I don't know why vertx doesn't close automatically. Recently the database has been prompted: 
Can't create more than max_prepared_stmt_count statements (current value: 16382)

**Comment by yiming33** [comment]
Every time we set the max_prepared_stmt_count value to be larger, but this is not a reasonable solution.

**Comment by codepitbull** [comment]
Thanks for the reproducer.
I nailed down the issue in the underlying driver.
https://github.com/mauricio/postgresql-async/issues/195

I can't do anything from here because I can't access the prepared statement from outside the driver.
This issue will never be resolved since Mauricio stopped working on the driver.

That's why we introduced the new Kotlin based driver which will replace the scaled based one. It will be part of the next release of Vert.x but you can already try it.
It's API compatible and you should be able to simply swap them out.

We won't be able to fix this, so closing this issue.

**Comment by oshai** [comment]
I created an issue in jasync-sql: https://github.com/jasync-sql/jasync-sql/issues/82
You're welcome to follow it and comment there.

**Comment by oshai** [comment]
@yiming33 the fix is in jasync version 0.8.65 (still need vertex to update to that version and release it). As a workaround, you could also try to reduce the connection idle time because when the connection is closed all it's prepared statements are deallocated by the database.

**Comment by wkgcass** [comment]
My test shows that there won't be any problem for non-stitched-sql strings:

The prepared statement will be reused if the sql string is exactly the same, which is the recommended by most books or guides.

The created prepared statements are the same as the code, so the space complexity is always O(1), unless you stitches strings into a sql based on requests.

```
mysqld --version
mysqld  Ver 5.7.17 for osx10.12 on x86_64 (Homebrew)
```

PSVM.java

```java
import io.vertx.core.Vertx;
import io.vertx.core.json.JsonArray;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.asyncsql.AsyncSQLClient;
import io.vertx.ext.asyncsql.MySQLClient;

public class PSVM {
    // static String sql = "insert into test_tbl(a,b) values (?,?)";
    static String sql = "select * from test_tbl where a = ? and b = ?";
    static int i;

    public static void main(String[] args) throws Exception {
        AsyncSQLClient cli = MySQLClient.createShared(Vertx.vertx(), new JsonObject()
            .put("host", "localhost")
            .put("port", 3306)
            .put("username", "root")
            .put("password", (String) null)
            .put("database", "test"));
        run(cli);
    }

    private static void run(AsyncSQLClient cli) {
        ++i;
        cli.queryWithParams(sql, new JsonArray().add(1).add(2), r -> {
            System.out.println(r.succeeded() + " " + i);
            if (r.failed()) {
                r.cause().printStackTrace();
            }
            run(cli);
        });
    }
}
```

run sql to check:

```
SHOW GLOBAL STATUS LIKE 'com_stmt%';

Com_stmt_execute	11301     # this value keeps increasing
Com_stmt_close	0
Com_stmt_fetch	0
Com_stmt_prepare	1
Com_stmt_reset	0
Com_stmt_send_long_data	0
Com_stmt_reprepare	0
```

and

```
select count(*) from performance_schema.prepared_statements_instances;

1
```

The program doesn't seem to run into any error (the field `i` keeps increasing).
